src/app-layer-parser.c

```diff
@@ -1149,6 +1149,17 @@ int AppLayerParserSetTxDetectState(const Flow *f,
     SCReturnInt(r);
 }
 
+bool AppLayerParserForProto(AppProto alproto)
```

Comment:
Elsewhere we use
```c
    if (alproto != ALPROTO_UNKNOWN && !(AppLayerParserIsTxAware(alproto))) {
        SCLogNotice("%s logger not enabled: protocol %s is disabled",
            name, AppProtoToString(alproto));
        return -1;
    }
```
ignore the alproto check.

I have a patch for 6.0dev that renames `AppLayerParserTxAware` to `AppLayerParserIsEnabled`

Reply:
Thanks, changed to use this. Right now trying to keep it simple to backport.